### PR TITLE
Expose "requireSession" in the requests and "pollTime" in the polling…

### PR DESCRIPTION
**Pull request by nahuelwolox**
## Summary ##

Expose the `requireSession` parameter in the requests so the user can modify it. It defaults to `true` as it was before.
Also exposed the `pollTime` parameter in `pollingRequest` so the user can modify the time between polls. It defaults to `1.0` as it was before.

### CODE REVIEW

**Comment by pablog** [comment]
Then you can remove `performAuthenticationRequest`, it's just a particular case of `performRequest` when `requiresSession = false`

**Comment by pablog** [comment]
I'd add all the functions combinations to the protocol when a parameter has a default value, to avoid breaking backwards compatibility:

```
/**
        Performs a request and returns a Signal producer.
        This function fails if no user is authenticated.
     
        - Parameters:
            - method: HTTP method for the request.
            - path: path to be appended to domain URL and subdomain URL.
            - parameters: request parameters.
            - headers: request headers. Authentication headers 
            are automatically injected. No need to be provided.
            - decoder: a closure of type Decoder
        - Returns:
            A SignalProducer where its value is the decoded entity and its
            error a RepositoryError.
     */
    func performRequest<T>(
        method: NetworkingMethod,
        path: String,
        parameters: [String: Any]?,
        headers: [String: String]?,
        requiresSession: Bool,
        decoder: @escaping Decoder<T>) -> SignalProducer<T, RepositoryError>

    func performRequest<T>(
        method: NetworkingMethod,
        path: String,
        headers: [String: String]?,
        requiresSession: Bool,
        decoder: @escaping Decoder<T>) -> SignalProducer<T, RepositoryError>

    func performRequest<T>(
        method: NetworkingMethod,
        path: String,
        parameters: [String: Any]?,
        requiresSession: Bool,
        decoder: @escaping Decoder<T>) -> SignalProducer<T, RepositoryError>

    func performRequest<T>(
        method: NetworkingMethod,
        path: String,
        parameters: [String: Any]?,
        headers: [String: String]?,
        decoder: @escaping Decoder<T>) -> SignalProducer<T, RepositoryError>

........
```

There are multiple combinations for each, so you can probably move the protocol to a new file.

**Comment by pablog** [comment]
Even, you can create a single protocol for each of the three functions we'll have in the implementation, create a new protocol (I'd keep current name) that extends the three protocols:

`protocol RepositoryType: PerformRequestRepositoryType, PerformPollingRequestRepositoryType, PerformRawRequestRepositoryType { }`

(Choose different names if you want).

**Comment by cristianames92** [comment]
Also, you can create a project card for this feature

**Comment by nahuelwolox** [comment]
I'll add a card in the project to do that refactor because I don't have time right now and it seems better to merge this as is and do the refactor when someone is free.
On that note, writing every possible combination seems very cumbersome.. Maybe we can create an object like `RequestConfiguration` that has every parameter created as the default values and just set the ones you need.

File: Networking.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -548,12 +503,6 @@
 			mainGroup = E1FCED7D1E31358D00E3408F;
 			productRefGroup = E1FCED871E31358D00E3408F /* Products */;
 			projectDirPath = "";
-			projectReferences = (
```
[the remark this answers is not in the document]
Comment (nahuelwolox) [reply Networking.xcodeproj/project.pbxproj]: I rebased from `swift-4-support` to be able to merge, all these changes come from that

[the remark this answers is not in the document]
Comment (pablog) [reply Networking.xcodeproj/project.pbxproj]: Those changes are made in this branch, not related with base branch, whatever it is.

[the remark this answers is not in the document]
Comment (nahuelwolox) [reply Networking.xcodeproj/project.pbxproj]: Sorry, I thought those were new changes. I checked and the modifications in the `.pbxproject` come from reverting the commit that removed dependencies from the cartfile and then modified the project accordingly. This wasn't needed in the first place so what you are seeing is a lot of undoing of those changes.

[the remark this answers is not in the document]
Comment (nahuelwolox) [reply Networking.xcodeproj/project.pbxproj]: It seems the rebase I did wasn't done correctly or something because I'm seeing things in the cartfile that shouldn't be there. I'll commit a fix for all of this.

**Comment by nahuelwolox** [comment]
I also deleted two tests that were outdated and failing in the base branch. I checked this with @cristianames92 and @jcaracciolo.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 80.84


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-networking-ios/branches/expose-request-config) for more details.

File: NetworkingTests/Tests/SessionManagerSpec.swift [hunk 2]
```diff
@@ -288,11 +288,6 @@ internal class SessionManagerSpec: QuickSpec {
                     sessionManager.bootstrap()
                 }
                 
-                it("sends no session signal") {
```
Comment (pablog) [inline NetworkingTests/Tests/SessionManagerSpec.swift]: This test should be kept, why did you remove it?

File: NetworkingTests/Tests/SessionManagerSpec.swift [hunk 3]
```diff
@@ -311,13 +306,6 @@ internal class SessionManagerSpec: QuickSpec {
                     expect(sessionManager.currentUser!.sessionToken).to(equal(updatedSessionToken))
                 }
                 
-                it("does not update the session token") {
```
Comment (pablog) [inline NetworkingTests/Tests/SessionManagerSpec.swift]: The same I think.

Comment (nahuelwolox) [reply NetworkingTests/Tests/SessionManagerSpec.swift]: @jcaracciolo modified that method in another PR to save the session token, so this test fails. As far as he told me he talked this with you and you were okay with removing both tests, but he forgot to remove them in that PR

File: NetworkingTests/Tests/SessionManagerSpec.swift (hunk 2 above)
Comment (nahuelwolox) [reply NetworkingTests/Tests/SessionManagerSpec.swift]: I meant the other test but he told me he talked this one with you too

**Comment by pablog** [comment]
@nahuelwolox OK, I see it. 

There are several methodological issues is all this working that ends up in different problems:

1) You changed functionality (added `saveSessionToken(user: user)` to `login(user: AuthenticableUser)`) in a branch called "swift-4-support".

2) You changed said functionality and didn't update the tests. Fix the tests in THAT branch, and then rebase this.

Also, update the comment in the modified function. @jcaracciolo

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 80.84
:x: Code Quality (Swiftlint): 84.0 - Minimum: 90.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-networking-ios/branches/expose-request-config) for more details.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 80.84
:x: Code Quality (Swiftlint): 84.0 - Minimum: 90.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-networking-ios/branches/expose-request-config) for more details.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 80.84
:x: Code Quality (Swiftlint): 84.0 - Minimum: 90.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-networking-ios/branches/expose-request-config) for more details.
